src/commands/infra/generate.ts

```diff
@@ -146,14 +147,22 @@ export const createGenerated = (projectPath: string): void => {
   logger.info(`Created generated directory: ${projectPath}`);
 };
 
-export const gitFetchPull = async (
+export const gitPull = async (
   sourcePath: string,
   safeLoggingUrl: string
 ): Promise<void> => {
   // Make sure we have the latest version of all releases cached locally
-  await simpleGit(sourcePath).fetch("all");
-  await simpleGit(sourcePath).pull("origin", "master");
-  logger.info(`${safeLoggingUrl} already cloned. Performing 'git pull'...`);
+  try {
+    await exec("git", ["symbolic-ref", "HEAD"], { cwd: sourcePath });
+    logger.info(
+      `${safeLoggingUrl} already cloned and a git branch is currently checked out. Performing 'git pull'...`
+    );
+    await simpleGit(sourcePath).pull();
+  } catch (err) {
+    logger.info(
```

Comment:
question: what is the assumption here. Is it that exec or simpleGit functions fail and we ignore the error?

Reply:
The logic is that if the exec command fails, then the passed in version argument is a git tag as opposed to a git branch, in which case don't perform `git pull` because it will cause the error that is seen now.